meta-resin-common/recipes-connectivity/usb-modeswitch/usb-modeswitch_2.2.5.bb

```diff
@@ -0,0 +1,29 @@
+# This package is a requirement for 3G USB modem support.
+#
+# Use this particular version, not the one shipped with openembedded,
+# because Ubuntu uses this version and we wish to use an Ubuntu patch.
```

Comment:
agherzan 21 hours ago
There is a hidden problem with this.

Basically you assume that the build system will take that specific 2.2.5 recipe which can get us into troubles if:

oe updates their recipe to 2.2.5 - this will depend on BBFILE_PRIORITY which happens to be fine
oe updates their recipe >= 2.2.5 - build system will pick up their recipe making our changes silently disappear.
So what we can do is:

use a PREFERRED VERSION on 2.2.5 based on the fact that given the BBFILE_PRIORITY, our recipe will still be used even if the PV is the same.
use a custom recipe which would provide usb-modeswitch, and our distro will set the default provider on this new provider
Additional comments:

is simpler (but is based on the assumption that BBFILE_PRIORITY of meta-resin > BBFILE_PRIORITY of meta-oe (which should be fine becuase curently it's the case)
is more complex but it doesn't depend on BBFILE_PRIORITY

I think we agreed with going with number 1